graphistry/umap_utils.py

```diff
@@ -286,17 +286,15 @@ def _bundle_embedding(self, emb, index):
         if emb.shape[1] == 2 and 'cudf.core.dataframe' not in str(getmodule(emb)):
             emb = pd.DataFrame(emb, columns=[config.X, config.Y], index=index)
         elif emb.shape[1] == 2 and 'cudf.core.dataframe' in str(getmodule(emb)):
-            import cudf
-            emb = cudf.DataFrame(emb.values, columns=[config.X, config.Y], index=index.values)
+            emb.rename(columns={0:config.X,1: config.Y},inplace=True)
         else:
             columns = [config.X, config.Y] + [
                 f"umap_{k}" for k in range(2, emb.shape[1] - 2)
             ]
             if 'cudf.core.dataframe' not in str(getmodule(emb)):
                 emb = pd.DataFrame(emb, columns=columns, index=index)
             elif 'cudf.core.dataframe' in str(getmodule(emb)):
-                import cudf
-                emb = cudf.DataFrame(emb.values, columns=columns, index=index.values)
+                emb.columns=columns
```

Comment:
why does shape matter? maybe 286-300 can all be `emb.columns=columns` ?

i'm pretty lost on cases here, and feels like there's happy-path coding here that may merit unit tests for 1/2/3/4-dim to ensure we're agreed on intended output col names

Reply:
It takes care of building an arbitrary umap embedding. I can set n_components=10 and still plot, send EMB to torch, etc

Reply:
yes, if `emb` is a df, I don't know why any of this munging is here, is all this just a df.rename(columns={..})?

Reply:
(earlier there was even weirder numpy & .values stuff)